src/main/java/htsjdk/variant/vcf/VCFSampleHeaderLine.java

```diff
@@ -1,13 +1,42 @@
 package htsjdk.variant.vcf;
 
+import java.util.Map;
+import java.util.Optional;
+
 /**
- * A class representing SAMPLE fields in the VCF header
  */
 public class VCFSampleHeaderLine extends VCFSimpleHeaderLine {
+
     private static final long serialVersionUID = 1L;
 
     public VCFSampleHeaderLine(String line, VCFHeaderVersion version) {
-        super(VCFConstants.SAMPLE_HEADER_KEY, VCFHeaderLineTranslator.parseLine(version, line, null));
+        // We need to use the V4 parser directly, since the V3 parser requires ALL permissible/expected
+        // tags to be supplied, which is inconsistent with modern structured header lines that allow
+        // other tags. So let validateForVersion detect any version incompatibility, ie., if this is ever
+        // called with a V3 version.
+        super(VCFConstants.SAMPLE_HEADER_KEY, new VCF4Parser().parseLine(line, expectedTagOrder));
+        validateForVersionOrThrow(version);
+    }
+
+    public VCFSampleHeaderLine(final Map<String, String> mapping) {
+        super(VCFConstants.SAMPLE_HEADER_KEY, mapping);
+    }
+
+    @Override
+    public Optional<VCFValidationFailure<VCFHeaderLine>> validateForVersion(final VCFHeaderVersion vcfTargetVersion) {
+        if (!vcfTargetVersion.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_0)) {
```

Comment:
You use the pattern here of either warning or returning an optional here.  It seems like it's done a number of times. 
It seem a bit weird, an extra layer of optionalness.  Should the decision to log a warning or not be left up to the caller?  If some options are only errors during strict validation maybe that should be recorded in the `ValidationFailure` object and then the caller can decide what to do with all of them.  If you don't like that idea, maybe pull out a method that either returns the option or warns and swallows it.  

I guess the reason behind it is probably because it's necessary since the super call might return a higher priority error, but that could be reworked so it's called first or the two are compared, or even both are returned.  

Not super important, but it seems awkward.